metrics/cli/test/deploy/deployer_test.dart

```diff
@@ -469,6 +477,33 @@ void main() {
       },
     );
 
+    test(
+      ".deploy() adds Firebase to the GCloud project with the created project id",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(false);
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+
+        await deployer.deploy();
+
+        verify(gcloudService.addFirebase(projectId)).called(once);
+      },
+    );
+
+    test(
+      ".deploy() deletes the temporary directory if GCloud service throws during the Firebase adding",
+      () async {
+        whenDirectoryExist().thenReturn(true);
+        whenPromptToSetupSentry().thenReturn(false);
+        whenCreateGCloudProject().thenAnswer((_) => Future.value(projectId));
+        whenAddFirebase().thenAnswer((_) => Future.error(stateError));
+
+        await expectLater(deployer.deploy(), throwsStateError);
+
+        verify(directory.deleteSync(recursive: true)).called(once);
+      },
+    );
+
     test(
       ".deploy() configures organization for the GCloud project",
```

Comment:
Should we test that this method deletes the temp directory in case the organization configuration fails?